core/src/main/java/org/elasticsearch/rest/action/search/RestSearchAction.java

```diff
@@ -130,13 +130,21 @@ private static void parseSearchSource(final SearchSourceBuilder searchSourceBuil
             searchSourceBuilder.query(queryBuilder);
         }
 
-        int from = request.paramAsInt("from", -1);
-        if (from != -1) {
-            searchSourceBuilder.from(from);
+//        int from = request.paramAsInt("from", -1);
+//        if (from != -1) {
+//            searchSourceBuilder.from(from);
+//        }
+//        int size = request.paramAsInt("size", -1);
+//        if (size != -1) {
+//            searchSourceBuilder.size(size);s
+//        }
+
+        if (request.hasParam("from")) {
+            searchSourceBuilder.from(request.paramAsInt("from", -1));
         }
-        int size = request.paramAsInt("size", -1);
```

Comment:
Maybe make the negative 1 a constant as opposed to the magic number that it is currently.